lib/elixir/lib/string.ex

```diff
@@ -813,6 +813,83 @@ defmodule String do
     rstrip(lstrip(string, char), char)
   end
 
+  @spec trim_leading(t, t) :: t
+  def trim_leading(string, to_strip) do
+    replace_leading(string, to_strip, "")
+  end
+
+  @spec trim_trailing(t, t) :: t
+  def trim_trailing(string, to_strip) do
+    replace_trailing(string, to_strip, "")
+  end
+
+  @spec trim(t, t) :: t
+  def trim(string, to_strip) do
+    string
+    |> trim_leading(to_strip)
+    |> trim_trailing(to_strip)
+  end
+
+  @spec pad_leading(t, non_neg_integer, t | [t]) :: t
+  def pad_leading(string, count, padding \\ [" "])
+
+  def pad_leading(string, count, padding) when is_binary(padding) do
+    pad(:leading, string, count, graphemes(padding))
+  end
+
+  def pad_leading(string, count, [_ | _] = padding)
+      when is_integer(count) and count >= 0 do
+    pad(:leading, string, count, padding)
+  end
+
+  @spec pad_trailing(t, non_neg_integer, t | [t]) :: t
+  def pad_trailing(string, count, padding \\ [" "])
+
+  def pad_trailing(string, count, padding) when is_binary(padding) do
+    pad(:trailing, string, count, graphemes(padding))
+  end
+
+  def pad_trailing(string, count, [_ | _] = padding)
+      when is_integer(count) and count >= 0 do
+    pad(:trailing, string, count, padding)
+  end
+
+  defp pad(kind, string, count, padding) do
+    string_len = length(string)
+    if string_len >= count do
```

Comment:
nvm, saw that you use the length below. I think the compiler doesn't optimize away the extra `length/1` call?